includes/validate.php

```diff
@@ -184,65 +184,47 @@ function edac_validate( $post_ID, $post, $action ) {
 function edac_remove_corrected_posts( $post_ID, $type, $pre = 1, $ruleset = 'php' ) {
 	global $wpdb;
 
-	$rules        = edac_register_rules();
-	$js_rule_ids  = array();
-	$php_rule_ids = array();
+	$rules      = edac_register_rules();
+	$rule_slugs = array();
 	foreach ( $rules as $rule ) {
-		if ( array_key_exists( 'ruleset', $rule ) && 'js' === $rule['ruleset'] ) {
-			$js_rule_ids[] = $rule['slug'];
-		} else {
-			$php_rule_ids[] = $rule['slug'];
+		if ( 'js' === $ruleset && isset( $rule['ruleset'] ) && 'js' === $rule['ruleset'] ) {
+			$rule_slugs[] = $rule['slug'];
+		} elseif ( 'js' !== $ruleset ) {
+			$rule_slugs[] = $rule['slug'];
 		}
 	}
-	
-	// Build a sql sanitized list from an array
-	// See: https://stackoverflow.com/questions/10634058/wordpress-prepared-statement-with-in-condition .
-	$js_rule_ids = array_map(
-		function ( $v ) {
-			return "'" . esc_sql( $v ) . "'";
-		},
-		$js_rule_ids
-	);
-	$js_rule_ids = implode( ',', $js_rule_ids );
-
-	// Build a sql sanitized list from an array
-	// See: https://stackoverflow.com/questions/10634058/wordpress-prepared-statement-with-in-condition .
-	$php_rule_ids = array_map(
-		function ( $v ) {
-			return "'" . esc_sql( $v ) . "'";
-		},
-		$php_rule_ids
-	);
-	$php_rule_ids = implode( ',', $php_rule_ids );
-
```

Comment:
This is no longer needed. Sanitization etc now happens in the query below - which is properly prepared